cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SwitchesChain.java

```diff
@@ -31,232 +29,173 @@
-            vl.getConnectables().forEach(c -> {
-                if (isDiscarded(c)) {
-                    return;
-                }
-                c.getTerminals()
-                    .forEach(t -> {
-                        if (!((Terminal) t).getVoltageLevel().equals(vl)) {
-                            return;
-                        }
-                        connectables.computeIfAbsent(((Terminal) t).getBusBreakerView().getBus().getId(), k -> new ArrayList<>()).add(c);
-                    });
-            });
-        }
+        VoltageLevel.NodeBreakerView.TopologyTraverser traverser = (node1, sw, node2) -> {
+            if (sw == swTerminal) {
+                return TraverseResult.TERMINATE_PATH;
+            }
+            nodes.add(node2);
+            return TraverseResult.CONTINUE;
+        };
 
-        private static boolean isSameVoltageLevel(Terminal terminal, VoltageLevel vl) {
-            return terminal.getVoltageLevel().equals(vl);
-        }
+        voltageLevel.getNodeBreakerView().traverse(node, traverser);
+        return nodes;
+    }
 
-        private int getConnectablesNodeCount(String nodeId) {
-            if (connectables.containsKey(nodeId)) {
-                return connectables.get(nodeId).size();
-            }
-            return 0;
-        }
+    private static List<String> expandBusBranch(VoltageLevel voltageLevel, String configuredBusId, Switch swTerminal) {
+        List<String> configuredBuses = new ArrayList<>();
+        configuredBuses.add(configuredBusId);
 
-        private Connectable<?> getFirstConductingEquipment(String nodeId) {
-            if (connectables.containsKey(nodeId)) {
-                return connectables.get(nodeId).stream().filter(NodeConnectables::isConductingEquipment).findFirst().orElse(null);
+        VoltageLevel.BusBreakerView.TopologyTraverser traverser = (busId1, sw, busId2) -> {
+            if (sw == swTerminal) {
+                return TraverseResult.TERMINATE_PATH;
             }
-            return null;
-        }
+            configuredBuses.add(busId2);
+            return TraverseResult.CONTINUE;
+        };
 
-        private static boolean isDiscarded(Connectable<?> connectable) {
-            return connectable.getType() == IdentifiableType.BUSBAR_SECTION;
-        }
-
-        private static boolean isConductingEquipment(Connectable<?> connectable) {
-            return connectable.getType() == IdentifiableType.LINE
-                || connectable.getType() == IdentifiableType.DANGLING_LINE;
-        }
+        voltageLevel.getBusBreakerView().traverse(configuredBusId, traverser);
+        return configuredBuses;
     }
 
-    private static class Adjacency {
-
-        private final Map<String, List<String>> adjacencies;
+    private static Terminal uniqueTerminalAssociatedConductingEquipmentNodeBreaker(VoltageLevel vl, List<Integer> nodes) {
```

Comment:
to avoid duplicating that method, you could use only one:
`private static <T> Terminal uniqueTerminalAssociatedConductingEquipment(VoltageLevel vl, List<T> vertices, Function<Terminal, T> terminalToVertex)`, where terminalToVertex will be 
- either `t -> t.getBusBreakerView().getBus()` 
- or `t -> t.getNodeBreakerView().getNode()`

Reply:
Done